packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -8,6 +8,24 @@ import { InMemoryCache } from 'apollo-cache-inmemory';
 import { Provider as ReduxProvider } from 'react-redux';
 import { Router } from '@magento/peregrine';
 
+/**
+ * To improve initial load time, as soon as this module is `require`d in we create
+ * an apollo cache object - it doesn't depend on any component props.
+ * The tradeoff is that we may be creating an instance we don't end up needing.
+ */
+const preInstantiatedCache = new InMemoryCache();
+
+/**
+ * We intentionally do not wait for the async function persistCache to complete
+ * because that would negatively affect the initial page load.
+ *
+ * The tradeoff is that any queries that run before the cache is persisted may not be persisted.
+ */
+persistCache({
+    cache: preInstantiatedCache,
+    storage: window.localStorage
+});
```

Comment:
Very informative comments. I agree with the direction of the changes here; I think *not awaiting* `persistCache()` will work toward your goal of not blocking initial page load.

Have you considered moving the cache instantiation and this `persistCache()` call into a `useEffect()`, though? If the goal is to free up the main thread to render as quickly as possible, and if we're already okay with allowing that render to occur before the Apollo cache gets prepared, why not defer this work until [after first paint](https://reactjs.org/docs/hooks-reference.html#useeffect) by doing it as an effect?

Reply:
I'll try that - thanks!

Reply:
Actually, I don't think I can do this.

>  if we're already okay with allowing that render to occur before the Apollo cache gets prepared

We're okay with the render occurring before the cache is _persisted_ (via `persistCache`) but we can't render before the cache is _created_ - `cache` is a required argument to the `ApolloClient` constructor and the `ApolloContext.Provider` and `ApolloProvider` below need an Apollo client instance on initial render.